app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt

```diff
@@ -0,0 +1,933 @@
+        LaunchedEffect(key1 = Unit) {
+            viewModel.loadAccounts(accountType)
+        }
+        AccountsShareScreen(
+            uiState = uiState,
+            isSearching = isSearching,
+            isFiltered = isFiltered,
+            navigateBack = { navigateBack },
+            onRetry = { viewModel.loadAccounts(accountType) },
+            isRefreshing = isRefreshing,
+            onRefresh = { viewModel.refresh(accountType) },
+            getUpdatedSearchList = { accountsList -> viewModel.searchInSharesList(accountsList, searchQuery)!! },
+            getUpdatedFilterList = { accountsList -> getFilterShareAccountList(accountsList = accountsList, filterList, viewModel, context) },
+            onItemClick = onItemClick
+        )
+    }
+}
+
+fun getFilterLoanAccountList(
+    accountsList: List<LoanAccount?>,
+    filterList: List<CheckboxStatus>,
+    viewModel: AccountsViewModel,
+    context: Context
+): List<LoanAccount?> {
+    val newList : MutableList<LoanAccount?> = mutableListOf()
+    for( filter in filterList)
+    {
+        if(filter.isChecked)
+            newList.addAll( viewModel.getFilteredLoanAccount(accountsList,filter, getFilterStrings(context))!! )
+    }
+    return newList
+}
+
+fun getFilterSavingsAccountList(
+    accountsList: List<SavingAccount?>,
+    filterList: List<CheckboxStatus>,
+    viewModel: AccountsViewModel,
+    context: Context
+): List<SavingAccount?> {
+
+    val newList : MutableList<SavingAccount?> = mutableListOf()
+    for( filter in filterList)
+    {
+        if( filter.isChecked )
+            newList.addAll( viewModel.getFilteredSavingsAccount(accountsList,filter, getFilterStrings(context))!! )
+    }
+    return newList
+}
+
+fun getFilterShareAccountList(
+    accountsList: List<ShareAccount?>,
+    filterList: List<CheckboxStatus>,
+    viewModel: AccountsViewModel,
+    context: Context
+): List<ShareAccount?> {
+    val newList : MutableList<ShareAccount?> = mutableListOf()
+    for( filter in filterList)
+    {
+        if(filter.isChecked)
+            newList.addAll( viewModel.getFilteredShareAccount(accountsList,filter, getFilterStrings(context))!! )
+    }
+    return newList
+}
+
+private fun getFilterStrings( context : Context?): AccountsFilterUtil {
+    return AccountsFilterUtil().apply {
+        this.activeString = context?.getString(R.string.active)
+        this.approvedString = context?.getString(R.string.approved)
+        this.approvalPendingString = context?.getString(R.string.approval_pending)
+        this.maturedString = context?.getString(R.string.matured)
+        this.waitingForDisburseString = context?.getString(R.string.waiting_for_disburse)
+        this.overpaidString = context?.getString(R.string.overpaid)
+        this.closedString = context?.getString(R.string.closed)
+        this.withdrawnString = context?.getString(R.string.withdrawn)
+        this.inArrearsString = context?.getString(R.string.in_arrears)
+    }
+}
```

Comment:
see how I have done in SavingsAccountTransactionScreen
SavingsTransactionCheckBoxFilter